compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Utils/AMDAIETransactionBuilder.cpp

```diff
@@ -67,6 +67,123 @@ LogicalResult TransactionBuilder::appendAddressPatch(uint32_t addr,
   return configureCustomTxnOp(deviceModel, opCode, data, size);
 }
 
+LogicalResult TransactionBuilder::appendLockOp(AMDAIE::LockOp lockOp) {
+  auto tile = lockOp.getTile().getDefiningOp<AMDAIE::TileOp>();
+  std::optional<int64_t> col = getConstantIntValue(tile.getCol());
+  std::optional<int64_t> row = getConstantIntValue(tile.getRow());
+  if (!col || !row) {
+    return tile->emitOpError()
+           << "expected column and row integer value/constant";
+  }
+  XAie_LocType tileLoc = XAie_TileLoc(*col, *row);
+  Lock lock{tileLoc, static_cast<uint8_t>(lockOp.getValue()),
+            static_cast<int8_t>(*lockOp.getInitValue())};
+  if (failed(initializeLock(deviceModel, lock))) return failure();
+  return success();
+}
+
+LogicalResult TransactionBuilder::appendDmaStartOp(
+    AMDAIE::DMAStartOp dmaStartOp) {
+  // Configure DMA Locks.
+  auto tile = dmaStartOp.getTile().getDefiningOp<AMDAIE::TileOp>();
+  std::optional<int64_t> col = getConstantIntValue(tile.getCol());
+  std::optional<int64_t> row = getConstantIntValue(tile.getRow());
+  if (!col || !row) {
+    return tile->emitOpError()
+           << "expected column and row integer value/constant";
+  }
+  XAie_LocType tileLoc = XAie_TileLoc(*col, *row);
+  FailureOr<XAie_DmaDesc> dmaDesc = initDMADesc(deviceModel, tileLoc);
+  if (failed(dmaDesc)) return failure();
+  // Configure DMA BD ops within DMA Start op.
+  dmaStartOp->walk([&](AMDAIE::DMABDOp dmaBdOp) -> WalkResult {
```

Comment:
You currently only support a dma_start with a single dma_bd op right? Then we should fail if it contains more than one. And I think you don't need a walk and could use `region.getOps<AMDAIE::DMABDOp>()` instead and then check that it only contains one element.